src/Database/Concerns/PendingScope.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Database\Concerns;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Scope;
+
+class PendingScope implements Scope
+{
+    /**
+     * All of the extensions to be added to the builder.
+     *
+     * @var string[]
+     */
+    protected $extensions = ['WithPending', 'WithoutPending', 'OnlyPending'];
+
+    /**
+     * Apply the scope to a given Eloquent query builder.
+     *
+     * @return void
+     */
+    public function apply(Builder $builder, Model $model)
+    {
+        $builder->when(! config('tenancy.pending.include_in_queries'), function (Builder $builder) {
+            $builder->whereNull('data->pending_since');
```

Comment:
The `data->` is *potentially* problematic because someone may want to:
1. rename the data column
2. create a custom column

Reply:
We should probably check for the data column name, and check if it's in the custom columns.

But we probably don't want to hardcode that here. So it'd be good to have an abstraction, perhaps in the package that handles the virtual columns.

Maybe a method like `getColumnForQuery()`/`generateColumnName()`/something along those lines.

Reply:
I opened a PR that adds the method to the package https://github.com/archtechx/virtualcolumn/pull/9

Reply:
@stancl I also updated the PendingScope to use the methods. I'll look for other uses of `data->` added in this PR and I'll also update that

Reply:
Okay, we can mark this as resolved then